### PR TITLE
Feat: instrument aiohttp with trace_configs argument 

### LINKED ISSUE #1073: Inject trace_configs as an argument to opentelemetry aiohttp_client._instrument instrumentation
**Is your feature request related to a problem?**

I would like to add data to a span based on on_request_chunk_sent event. Using the request/response hooks doesn't provide a good solution for my use case since it's "too late" in the timeline. 

**Describe the solution you'd like**

I would like to pass trace_configs (or a single trace_config) to aiohttp_client opentelemetry instrumentation. It will allow me to specify which events I would like to "catch" and will allow more customization solving many use-cases that the request/response hooks cannot.

**Describe alternatives you've considered**

I've doubled the instrument_init method and set it up with my own trace_configs, bypassing opentelemetry implementation. 

**Additional context**
If this issue will be approved I would like to contribute my solution, thanks

### PR DESCRIPTION
**Pull request by nemoshlag**
# Description

Adding trace_configs argument to aiohttp client wrapper enables additional customization. For instance, on_request_chunk_sent event is called after span was created and before request was ended - making it impossible to capture data chunks when using request or response hooks.

Fixes #1073 (issue)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

The trace_configs feature is already integrated and tested with aiohttp wrapper library (https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-aiohttp-client/tests/test_aiohttp_client_integration.py). 
An additional test was added to verify new functionality - creating 2 spans with passing trace_configs as an argument to instrument func.

- [ ] Test A -> test_instrument_with_custom_trace_config 

# Does This PR Require a Core Repo Change?

- [ ] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [ ] Unit tests have been added

### CODE REVIEW

**Comment by sanketmehta28** [comment]
Please do resolve build errors

**Comment by sanketmehta28** [comment]
Please do resolve the build errors

**Comment by sanketmehta28** [comment]
still 9 checks are failing. @nemoshlag : Can you please look into that

File: instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py [hunk 1]
```diff
@@ -328,12 +327,15 @@ def _instrument(self, **kwargs):
...
                 ``response_hook``: An optional callback which is invoked right before the span is finished processing a response.
+                ``trace_configs``: An optional list of aiohttp.TraceConfig items, allowing customize enrichment of spans
+                 based on aiohttp events (see specification: https://docs.aiohttp.org/en/stable/tracing_reference.html)
         """
         _instrument(
             tracer_provider=kwargs.get("tracer_provider"),
             url_filter=kwargs.get("url_filter"),
             request_hook=kwargs.get("request_hook"),
             response_hook=kwargs.get("response_hook"),
+            trace_configs=list(kwargs.get("trace_configs")),
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py:338]:
```suggestion
            trace_configs=list(kwargs.get("trace_configs", [])),
```

File: instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py [hunk 2]
```diff
@@ -273,8 +274,6 @@ def instrumented_init(wrapped, instance, args, kwargs):
         if context_api.get_value(_SUPPRESS_INSTRUMENTATION_KEY):
             return wrapped(*args, **kwargs)
 
-        trace_configs = list(kwargs.get("trace_configs") or ())
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py:276]: Does this mean we are removing the ability to pass the `trace_configs` from `aiohttp.ClientSession.__init__` and adding it to `.instrument()`?

Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py:276]: Does this mean we are removing the ability to pass the `trace_configs` from `aiohttp.ClientSession.__init__` and adding it to `.instrument()`?

File: instrumentation/opentelemetry-instrumentation-aiohttp-client/tests/test_aiohttp_client_integration.py [hunk 3]
```diff
@@ -403,6 +403,19 @@ async def create_session(server: aiohttp.test_utils.TestServer):
         run_with_test_server(create_session, self.URL, self.default_handler)
         self.assert_spans(1)
 
+    def test_instrument_with_custom_trace_config(self):
+        AioHttpClientInstrumentor().uninstrument()
+        self.assert_spans(0)
+        AioHttpClientInstrumentor().instrument(
+            trace_config=aiohttp_client.create_trace_config()
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-aiohttp-client/tests/test_aiohttp_client_integration.py:410]:
```suggestion
            trace_configs=aiohttp_client.create_trace_config()
```

Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-aiohttp-client/tests/test_aiohttp_client_integration.py:410]:
```suggestion
            trace_configs=aiohttp_client.create_trace_config()
```
```suggestion
            trace_config=aiohttp_client.create_trace_config()
```

File: instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py (hunk 2 above)
Comment (nemoshlag) [reply instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py:276]: Hi @srikanthccv thanks! I see now removing it from ClientSession is problematic. Ideally, I would like to be able to pass trace_configs from both -> Does it make sense, or I'm working with the library the wrong way?

File: CHANGELOG.md [hunk 4]
```diff
@@ -17,6 +17,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 
 
+### Added
+- `opentelemetry-instrumentation-aiohttp-client` Add support for optional custom trace_configs argument.
+  ([1079](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1079))
```
Comment (srikanthccv) [inline CHANGELOG.md:22]: Move this `Changed` section in Unreleased.

File: instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py [hunk 5]
```diff
@@ -262,6 +262,7 @@ def _instrument(
     url_filter: _UrlFilterT = None,
     request_hook: _RequestHookT = None,
     response_hook: _ResponseHookT = None,
+    trace_configs: Collection = (),
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py:265]: Make this an optional list. You can't append to tuple.

File: instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py [hunk 6]
```diff
@@ -273,7 +274,9 @@ def instrumented_init(wrapped, instance, args, kwargs):
         if context_api.get_value(_SUPPRESS_INSTRUMENTATION_KEY):
             return wrapped(*args, **kwargs)
 
-        trace_configs = list(kwargs.get("trace_configs") or ())
+        trace_configs.append(kwargs.get("trace_configs")) if kwargs.get(
+            "trace_configs"
+        ) else None
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py:279]: This else None is unnecessary. You could just do `if x: y`

**Comment by srikanthccv** [comment]
@nemoshlag This is easy fix. Please address the comments and it should be ready to merge.

File: CHANGELOG.md (hunk 4 above)
Comment (srikanthccv) [reply CHANGELOG.md:47]: This is still not moved

File: instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py [hunk 7]
```diff
@@ -262,18 +262,24 @@ def _instrument(
     url_filter: _UrlFilterT = None,
     request_hook: _RequestHookT = None,
     response_hook: _ResponseHookT = None,
+    trace_configs: [aiohttp.TraceConfig] = None,
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py:265]: Is this `[aiohttp.TraceConfig]` valid? Shouldn't it be `Optional[aiohttp.TraceConfig]`?

File: CHANGELOG.md (hunk 4 above)
Comment (nemoshlag) [reply CHANGELOG.md:28]: Hi @srikanthccv, thanks, did you mean to place it here?

**Review by srikanthccv — APPROVED** [review]

**Pull request merged**